x/auth/ante/expected_keepers.go

```diff
@@ -19,6 +20,13 @@ type AccountKeeper interface {
 	AddressCodec() address.Codec
 }
 
+// UnorderedNonceManager defines the contract needed for UnorderedNonce management.
+type UnorderedNonceManager interface {
+	RemoveExpiredUnorderedNonces(ctx sdk.Context) error
+	AddUnorderedNonce(ctx sdk.Context, sender []byte, timestamp time.Time) error
+	ContainsUnorderedNonce(ctx sdk.Context, sender []byte, timestamp time.Time) (bool, error)
```

Comment:
Probably feasible to collapse this into a single TryAdd method - that's a bit safer, but this is okay. TryAdd is safer because Add should actually fail if the entry exists and we want to avoid calling Has twice

Reply:
made adjustment here: [TryAdd](https://github.com/cosmos/cosmos-sdk/pull/24010/commits/a4a40c21e34fa4134747db88eace520452746e0a)

makes the ante code simpler 👍